src/datasets/translation_datasets.py

```diff
@@ -1,15 +1,20 @@
 import numpy as np
+import torch
 from datasets import load_dataset
 from src.utils.common import clean_text
 from torch.utils.data import IterableDataset, Dataset
 from tqdm import tqdm
 
+from transformers import NllbTokenizer
+
 
 class NLLBDataset(IterableDataset):
     def __init__(
         self,
         probas: dict,
         k: float = 0.3,
+        tokenizer: NllbTokenizer = None,
+        max_seq_len: int = None,
```

Comment:
I think it is better to keep this class without any changes and provide additional `TokenizedDataset`. So, like with `SlicesDataset` we will wrap augmented or any other dataset in it. And inside this `TokenizedDataset` we will tokenize sequence and collate into a batch